lib/Dialect/HWArith/HWArithOps.cpp

```diff
@@ -18,6 +18,29 @@
 using namespace circt;
 using namespace hwarith;
 
+//===----------------------------------------------------------------------===//
+// CastOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult CastOp::verify() {
+  auto inType = in().getType();
+  auto outType = out().getType();
+  bool isInSignless = !isHWArithIntegerType(inType);
+  bool isOutSignless = !isHWArithIntegerType(outType);
+
+  if (isInSignless && isOutSignless)
+    return emitError("at least one type needs to be a HWArithIntergerType");
```

Comment:
`HWArithIntergerType` -> `HWArithIntegerType`.
Also, this is a user-facing error but the error message refers to the type constraint in the source code. Maybe it should just be "at least one type needs to carry sign semantics (ui/si)"